lib/pure/times.nim

```diff
@@ -943,31 +995,32 @@ else:
     # as a result of offset changes (normally due to dst)
     let utcUnix = adjTime.seconds + utcOffset
     let (finalOffset, dst) = getLocalOffsetAndDst(utcUnix)
-    result.adjTime = initTime(utcUnix - finalOffset, adjTime.nanosecond)
+    result.time = initTime(utcUnix, adjTime.nanosecond)
     result.utcOffset = finalOffset
     result.isDst = dst
 
-proc utcZoneInfoFromUtc(time: Time): ZonedTime =
-  result.adjTime = time
-  result.utcOffset = 0
-  result.isDst = false
+proc utcTzInfo(time: Time): ZonedTime =
+  ZonedTime(utcOffset: 0, isDst: false, time: time)
 
-proc utcZoneInfoFromTz(adjTime: Time): ZonedTime =
-  utcZoneInfoFromUtc(adjTime) # adjTime == time since we are in UTC
+var utcInstance = newTimezone("Etc/UTC", utcTzInfo, utcTzInfo)
+var localInstance = newTimezone("LOCAL",
+  localZonedTimeFromTime, localZonedTimeFromAdjTime)
 
 proc utc*(): TimeZone =
   ## Get the ``Timezone`` implementation for the UTC timezone.
   runnableExamples:
     doAssert now().utc.timezone == utc()
     doAssert utc().name == "Etc/UTC"
-  Timezone(zoneInfoFromUtc: utcZoneInfoFromUtc, zoneInfoFromTz: utcZoneInfoFromTz, name: "Etc/UTC")
+  {.gcsafe.}:
+    result = utcInstance
```

Comment:
Is this safe since `utcInstance` will never be deallocated or mutated? I could use `{.threadvar.}` instead but if this is safe I think it's better.

Reply:
It's not safe, the write barriers the GC uses are not threadsafe.

Reply:
OK, thread safety in Nim is still kind of a mystery to me :-) Changed it to use `{.threadvar.}`